### PR TITLE
Server-side rendering docs

### PR DESCRIPTION
**Pull request by matthewp**
This documents how to setup and use SSR features in Astro.

### CODE REVIEW

File: src/pages/en/reference/adapter-reference.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+---
+layout: ~/layouts/MainLayout.astro
+title: Astro Adapter API (experimental)
+---
+
+Astro is designed to make it easy to deploy to any cloud provider for SSR (server-side rendering). This ability is provided by __adapters__, which are [integrations](./integrations-reference/).
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:6]:
```suggestion
Astro is designed to make it easy to deploy to any cloud provider for SSR (server-side rendering). This ability is provided by __adapters__, which are [integrations](/en/reference/integrations-reference/).
```

File: src/pages/en/reference/adapter-reference.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+> Server-side rendering in Astro is *experimental*. If you are interested in building an adapter for a host now is the perfect time to help shape these APIs. If you are worried about breaking changes this might be a little too soon for you.
+
+## What is an adapter
+
+An adapter is a special kind of [integrations](./integrations-reference/) that provide an entrypoint for server-side rendering. An adapter does two things:
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:12]:
```suggestion
An adapter is a special kind of [integration](/en/reference/integrations-reference/) that provides an entrypoint for server-side rendering. An adapter does two things:
```

File: src/pages/en/reference/adapter-reference.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+- Implements host-specific APIs for handling requests.
+- Configures the build according to host conventions.
+
+## Building an Adapter
+
+An adapter is an [integration](./integrations-reference/) and can do anything that an integration can do.
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:19]:
```suggestion
An adapter is an [integration](/en/reference/integrations-reference/) and can do anything that an integration can do.
```

File: src/pages/en/reference/adapter-reference.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+The properties are:
+
+* __name__: A unique name for your adapter, used for logging.
+* __serverEntrypoint__: The entrypoint for server-side rendering.
+* __exports__: An array of named exports when used in conjunction with `createExports` (explained below).
+
+### Server Entrypoint
+
+Astro's adapter API attempts to work with any type of host, and gives a flexible way to confirm to the host APIs.
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:57]: "way to confirm to the host APIs."

(Sorry, I'm not familiar with the proper language to describe this, so I'm just checking that this is intentional wording. In plain English, I'd be expecting "to confirm "X" to the host APIs" or "to provide confirmation to the host APIs")

File: src/pages/en/reference/adapter-reference.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+Some serverless hosts expect you to export a function, such as `handler`:
+
+```js
+export function handler(event, context) {
+  // ...
+}
+```
+
+With the adapter API you achieve these by implementing `createExports` in your `serverEntrypoint`:
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:69]:
```suggestion
With the adapter API you achieve this by implementing `createExports` in your `serverEntrypoint`:
```

File: src/pages/en/reference/adapter-reference.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+  const handler = (event, context) => {
+    // ...
+  };
+
+  return { handler };
+}
+```
+
+And then in your integration, where you call `setAdapter` provide this name in `exports`:
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:85]:
```suggestion
And then in your integration, where you call `setAdapter`, provide this name in `exports`:
```

File: src/pages/en/reference/adapter-reference.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        });
+      },
+    },
+  };
+}
+```
+
+#### Start
+
+Some hosts expect you to *start* the server yourselves, for example by listening to a port. For these types of hosts, the adapter API allows you to export a `start` function which will be called when the bundle script is ran.
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:106]:
```suggestion
Some hosts expect you to *start* the server yourselves, for example by listening to a port. For these types of hosts, the adapter API allows you to export a `start` function which will be called when the bundle script is run.
```

File: src/pages/en/reference/adapter-reference.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+    );
+  });
+}
+```
+
+The following methods are provided:
+
+##### app.match(request)
+
+This method is used to determine a request is matched by the Astro app's routing rules.
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:143]:
```suggestion
This method is used to determine if a request is matched by the Astro app's routing rules.
```

File: src/pages/en/reference/adapter-reference.md (hunk 4 above)
Comment (matthewp) [reply src/pages/en/reference/adapter-reference.md:57]: Oops, I meant __conform__

Comment (sarah11918) [reply src/pages/en/reference/adapter-reference.md:57]: Ha! That was my next guess!  :)

File: src/pages/en/reference/adapter-reference.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    event.respondWith(
+      app.render(event.request)
+    );
+  });
+}
+```
+
+The following methods are provided:
+
+##### app.match(request)
```
Comment (sarah11918) [inline src/pages/en/reference/adapter-reference.md:141]: Just a thought that maybe this would flow better if these two were swapped in order? app.render seems to be the more common method, and the one used in the example above?

If the first thing you're going to say about app.match(request) is that you can usually use the other one (that you haven't described yet) instead, then maybe it makes sense to describe app.render first? And, right in there, you can say that it can be used for 404s if you provide a 404.astro file?

 Then, when you get to app.match, you've already talked about the more commonly used method, and you can just say to use this if you want to handle 404s differently?

Comment (matthewp) [reply src/pages/en/reference/adapter-reference.md:141]: Good point, I'll switch this.
